src/OmniSharp/Workers/Refactoring/FixUsings.cs

```diff
@@ -0,0 +1,240 @@
+                        {
+                            continue;
+                        }
+                        var usingActions = GetUsingActions(document, codeActionProvider, pointDiagnostics, "using");
+                        foreach (var action in usingActions)
+                        {
+                            var operations = await action.GetOperationsAsync(CancellationToken.None).ConfigureAwait(false);
+                            if (operations != null)
+                            {
+                                foreach (var codeOperation in operations)
+                                {
+                                    usingOperations.Add(action.Title, codeOperation);
+                                }
+                            }
+                        }
+
+                        if (usingOperations.Count() == 1)
+                        {
+                            //Only one operation - apply it
+                            usingOperations.Single().Value.Apply(workspace, CancellationToken.None);
+                            updated = true;
+                        }
+                        else if (usingOperations.Count() > 1)
+                        {
+                            if (usingOperations.Keys.Any(k => k.StartsWith("using Microsoft.CodeAnalysis.CSharp")))
+                            {
+                                //Resolve all usings which are common to csharp/visualbasic
+                                var csharpOperation = usingOperations.Keys.Where(k => k.StartsWith("using Microsoft.CodeAnalysis.CSharp")).First();
+                                usingOperations[csharpOperation].Apply(workspace, CancellationToken.None);
+                                updated = true;
+                            }
+                            else
+                            {
+                                //What is left is ambiguous
+                                ambiguousNodes.Add(node);
+                                var unresolvedText = node.Identifier.ValueText;
+                                var unresolvedLocation = node.GetLocation().GetLineSpan().StartLinePosition;
+                                ambiguous.Add(new QuickFix
+                                {
+                                    Line = unresolvedLocation.Line + 1,
+                                    Column = unresolvedLocation.Character + 1,
+                                    FileName = fileName,
+                                    Text = "`" + unresolvedText + "`" + " is ambiguous"
+                                });
+                            }
+                        }
+                    }
+                }
+
+                processMore = updated;
+            }
+
+            return ambiguous;
+        }
+
+        private static async Task RemoveDuplicateUsings(string fileName, OmnisharpWorkspace workspace, Document document, SemanticModel semanticModel, ICodeActionProvider codeActionProvider)
```

Comment:
This method should be renamed to RemoveUsings or something similar. It isn't just removing duplicates any more.